### PR TITLE
windows: Stop beeping

### LINKED ISSUE #14857: Annoying sounds
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The zed makes an annoying beep when a certain key is entered. Is there an option to turn it off

### Environment

win10

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

### PR DESCRIPTION
**Pull request by JunkuiZhang**
Close #14857 

Release Notes:

- N/A

### CODE REVIEW

File: crates/gpui/src/platform/windows/events.rs [hunk 1]
```diff
@@ -273,7 +274,8 @@ fn handle_syskeydown_msg(
         keystroke,
         is_held: lparam.0 & (0x1 << 30) > 0,
     };
-    let result = if func(PlatformInput::KeyDown(event)).default_prevented {
+    let result = if !func(PlatformInput::KeyDown(event)).propagate {
+        state_ptr.state.borrow_mut().system_key_handled = true;
```
Comment (JunkuiZhang) [inline crates/gpui/src/platform/windows/events.rs:278]: I'm not sure how heavy `RefCell` is. Should I use `AtomicBool` for this case instead?

Comment (mikayla-maki) [reply crates/gpui/src/platform/windows/events.rs:278]: RefCell should be a lot cheaper, as there's no atomic synchronization.

**Comment by mikayla-maki** [comment]
Thank you!

**Comment by JunkuiZhang** [comment]
> Thank you!

I'm glad to hear that you’ve recently recovered from COVID-19. I hope you’re feeling much better now. Please take care of yourself and be well :)

**Comment by mikayla-maki** [comment]
I appreciate the well wishes @JunkuiZhang :D

Also, I'd love to chat synchronously sometime about Zed if you're interested. Are there any western platforms that you're available on? We have a Discord, twitter, and could figure something out if there's another place :)

**Comment by JunkuiZhang** [comment]
> Also, I'd love to chat synchronously sometime about Zed if you're interested. Are there any western platforms that you're available on? We have a Discord, twitter, and could figure something out if there's another place :)

I have joined the Discord server, but since I need to use a VPN to access Discord, I am not always online. However, I will log in whenever you need me.

**Comment by mikayla-maki** [comment]
Oh lovely! Please @ or DM me sometime so I know your handle, I'd like to make sure we're in contact when we decide to start on a windows port; as you've basically built the whole thing yourself!

**Comment by JunkuiZhang** [comment]
> Oh lovely! Please @ or DM me sometime so I know your handle, I'd like to make sure we're in contact when we decide to start on a windows port; as you've basically built the whole thing yourself!

I just sent you a message. My Discord account name is zhangxiaobai. I am new to Discord, so I hope I did it correctly.

**Pull request merged**